logging/structured/src/main/scala/tofu/logging/location/LocationLogging.scala

```diff
@@ -0,0 +1,110 @@
+package tofu.logging.location
+
+import cats.effect.Sync
+import org.slf4j.{LoggerFactory, Marker}
+import tofu.compat.`package`.unused
+import tofu.logging.Logging.{Debug, Error, Info, Level, Trace, Warn}
+import tofu.logging.impl.{ContextSyncLoggingImpl, SyncLogging}
+import tofu.logging.{LoggableContext, LoggedValue, Logging, Logs}
+import tofu.syntax.monadic._
+import scala.reflect.macros.blackbox
+
+case class Location(className: String, line: Int) {
+  def loggerName: String = className
+}
+
+object Location {
+  implicit def instance: Location =
+    macro LocationMacro.impl
+
+  object LocationMacro {
+    def impl(c: blackbox.Context): c.Tree = {
+      import c.universe._
+      val file = c.enclosingPosition.source.path.replace("/", ".").dropRight(6).drop(1) //this is adhoc af and just for POC purpose
+      val line = c.enclosingPosition.line
+      q"_root_.tofu.logging.location.Location($file, $line)"
+    }
+  }
+}
+
+trait LocationLogging[F[_]] {
```

Comment:
I don't think we need another implicit bound for that. Too much migration pain for people who just want to add some functionality to their existing logs